src/NfMaxOrd/ResidueRing.jl

```diff
@@ -171,6 +173,24 @@ end
 
 *(x::NfMaxOrdQuoRingElem, y::fmpz) = y*x
 
+function ^(a::NfMaxOrdQuoRingElem, f::fmpz)
+  f==0 && return one(parent(a))
+  f==1 && return a
+  if f<0
+    f=-f
+    a = inv(a)
+  end
+  b = a^(div(f, 2))
+  b *= b
+  if isodd(f)
+    b *= a
+  end
+  return b
+end
+
+^(a::NfMaxOrdQuoRingElem, f::Integer) = a^fmpz(f)
```

Comment:
Verstehe ich nicht. Warum hat `^(x, p::Integer)` Vorrang? Die Signatur hier ist spezieller als die Signatur in Base.